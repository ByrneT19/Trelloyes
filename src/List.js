import React from 'react';
import './List.css';
import Card from './Card';

List.defaultProps = { 
    header: '', 
    card: [] }

export default function List(props) {
   
    return (
        <div>
            <section className="List">
                <header className="List-header">
                    <h2>{props.header}</h2>
                </header>
                <div className="List-cards">
                    {props.card.map((card) =>
                    <Card 
                        key={card.id}
                        title={card.title}
                        content={card.content}
                    />
                    )}
                    <div>
                        <button
                        type='button'
                        className='List-add-button'
                        >
                            + Add Random Card
                        </button>
                    </div>
                </div>
            </section>
        </div>
    )
}

