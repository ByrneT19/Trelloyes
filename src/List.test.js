import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
// import STORE from './store';

describe('List component', () => {
    it('it renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Renders the UI as expected', () => {
        const tree = renderer
            .create(<List 
                header={''}
                card={[]}
                />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})