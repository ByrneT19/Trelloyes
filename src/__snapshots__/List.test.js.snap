// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component Renders the UI as expected 1`] = `
<div>
  <section
    className="List"
  >
    <header
      className="List-header"
    >
      <h2>
        
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <button
          className="List-add-button"
          type="button"
        >
          + Add Random Card
        </button>
      </div>
    </div>
  </section>
</div>
`;
